File: JAVA/9.palindrome number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return (false);
        } else {
            int rev = 0, r, t = x;
            while (t != 0) {
                r = t % 10;
                rev = (rev * 10) + r;
                t = t / 10;
            }
            if (rev == x) {
                return (true);
            } else {
                return (false);
            }
        }

    }
}

File: C++/9. Palindrome Number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        long int n, s;
        n = x;
        if (n == 0)
            return true;
        else if (n < 0)
            return false;
        else {
            while (n != 0) {
                s = (s * 10) + (n % 10);
                n = n / 10;
            }
        }
        if (s == x)
            return true;
        else
            return false;
    }
};